### DISCUSSION TITLE
Is there Python version of it?

### DISCUSSION THREAD

**Discussion by LeeDongGeon1996** [DaveSkender/Stock.Indicators#393] [Q&A] [answered]
Is there Python version of it? 
I've compiled DaveSkender/Stock.Indicators as DLL and using it in Python.
But to use it perfectly in Python, some of C# module needs to be imported and little more of code for conversion is needed when using in Python. 

Until now, I've written some Python code only for `get_Sma()`. If there is still no Python conversion of the project, I have willing to write more code for it.

I would like to know how do you think about this.
Best regards,
DongGeon Lee

**Comment by DaveSkender** [comment, marked as answer]
We've not built a separate Python version or wrapper yet, but are aware of integrations, like [Python.NET](https://pypi.org/project/pythonnet), that can be used to incorporate this library into Python projects.  We've also built this library using the [Common Language Specification (CLS)](https://docs.microsoft.com/en-us/dotnet/standard/common-type-system) so that it can be used in a broad set of languages.

I like the idea of providing a Python wrapper, given the popularity of Python, but have not done that yet.  We'd probably not do a full re-write in native Python code.  With that said, I would consider doing the wrapper to create a community Python package if we can get a strong Python contributor to lead that effort (#394).

In any case, I welcome any feedback or discussion on how we could provide better Python support, including interests in contributing to this project.

**Reply by LeeDongGeon1996** [reply]
Yes, what I would like to talk about is the "wrapper". 

Actually, I'm working on the project, "Stock Trading with Reinforcement Learning", needs to get historical stock indicator, and finally found DaveSkender/Stock.Indicators, which is perfect to us. But the only problem was it is written in C# and we wanted to use it in Python. That's why I wrote the code for conversion of it.

I also used [Python.NET](https://pypi.org/project/pythonnet) you mentioned above and successfully wrote the wrapper for `get_Sma()`. Now, I think I can write the wrapper for the rest part of project.

Though I cannot tell you I'm the person who can write awesome code, having perfect architecture and efficiency because I'm still student and there are so many "pro"s in the world, I think I can wrote kind of good conversion of this project based on my experience. 
So, If you don't mind and it is not urgent, I would like to try to contribute on it for conversioning.
Would like to hear your thought.

Sincerely,
DongGeon Lee

**Reply by DaveSkender** [reply]
We'd love to have your help in contributing to this project.  If interested in contributing to the Python wrapper, let's continue the discussion in #394 to coordinate implementation details.  I could definitely use some help on this one.  Not to worry if you're a student, I'm newbie-friendly -- we'll do incremental code reviews to ensure everything is to a production-level standard.

**Reply by LeeDongGeon1996** [reply]
Okay :)

**Comment by DaveSkender** [comment]
As a side note, if you're compiling the DLL yourself from source code, be sure to use the `Release` configuration since it's up to 2× faster than the `Debug` version.

If you're not sure how to do that, the easier way to get a `Release` version of the DLL is to:

1. Download the latest package from the [NuGet](https://www.nuget.org/packages/Skender.Stock.Indicators) site
2. Change the file extension from `.nupkg` to `.zip` to open the archive
3. Pick the `lib` folder of the most modern framework that you can support (`net5.0` is currently the latest)

The newer .NET 5 and .NET Core variants are notably faster than the older .NET Standard and .NET Framework variants.

**Reply by LeeDongGeon1996** [reply]
Oh.. thank you for letting me know about this. Actually, I was really struggling about building DLL of this project because I'm not familiar with `.net`. So I compiled using  `dotnet build` command to build DLL and the compiled DLL works correctly in my Python code. Now, I will try again with following your instruction.
